android/src/main/java/com/airbnb/android/react/maps/AirMapView.java

```diff
@@ -289,6 +296,24 @@ private boolean hasPermissions() {
                 checkSelfPermission(getContext(), PERMISSIONS[1]) == PackageManager.PERMISSION_GRANTED;
     }
 
+    /*
+    onDestroy is final method so I can't override it.
+     */
+    public synchronized  void doDestroy() {
+        if (lifecycleListener != null && context != null) {
```

Comment:
What happens if this gets called while the map is initializing before onMapReady is called? Can you cancel the initialization? Maybe return early from onMapReady if destroyed?